### PR TITLE
[Action] Build & Deploy

### PR DESCRIPTION
**Pull request by Yserz**

### CODE REVIEW

File: .github/workflows/build_deploy.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        uses: actions/checkout@v2
+
+      - name: Set BRANCH_NAME environment variable
+        run: echo "::set-env name=BRANCH_NAME::$(git branch --show-current)"
+
+      - name: Set TAG environment variable
+        run: echo "::set-env name=TAG::$(git tag --points-at HEAD)"
+
+      - name: Print env vars
+        run: echo -e "GITHUB_SHA = $GITHUB_SHA , TAG = $TAG , BRANCH_NAME = $BRANCH_NAME , GITHUB_REF = $GITHUB_REF"
```
Comment (ffflorian) [inline .github/workflows/build_deploy.yml:32]:
```suggestion
        run: echo -e "GITHUB_SHA = ${GITHUB_SHA}, TAG = ${TAG}, BRANCH_NAME = ${BRANCH_NAME}, GITHUB_REF = ${GITHUB_REF}"
```

File: .github/workflows/build_deploy.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      DOCKER_PASSWORD: ${{secrets.DOCKER_PASSWORD}}
+      DOCKER_USERNAME: ${{secrets.DOCKER_USERNAME}}
+      DISTRIBUTION: ${{secrets[matrix.DISTRIBUTION]}}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set BRANCH_NAME environment variable
+        run: echo "::set-env name=BRANCH_NAME::$(git branch --show-current)"
```
Comment (ffflorian) [inline .github/workflows/build_deploy.yml:26]: Why not use `${GITHUB_REF##*/}`?

**Review by ffflorian — CHANGES_REQUESTED** [review]

File: .github/workflows/build_deploy.yml (hunk 2 above)
Comment (Yserz) [reply .github/workflows/build_deploy.yml:26]: GITHUB_REF is not only the branch, especially for pull requests this expression is wrong.

Comment (ffflorian) [reply .github/workflows/build_deploy.yml:26]: But you are using this on a push, not a pull request.

Comment (Yserz) [reply .github/workflows/build_deploy.yml:26]: Yep, but for testing I set this to pull request as well. So it's better for debugging and resistant to changes.

Comment (Yserz) [reply .github/workflows/build_deploy.yml:26]: Also it can be copied to other workflows without changes.

**Pull request merged**